[Leon Tan] iP

leontan2 opened the pull request:
# The Bob Chatbot
> "It is what it is." - Bob

## Bob frees your mind of having to remember things you need to do. It's,
- Text-based 
- Easy to learn (not really)
- ~~FAST~~ SUPER FAST to use (not really)

## All you need to do is,
1. Be in NUS CS
1. Have a computer :upside_down_face:
1. Download it from [here](https://github.com/nus-cs2103-AY2324S1).
1. Run the jar file.

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Also, here's my `main` method.
```java
public static void main(String[] args) {
    new Duke("data/tasks.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

leontan2 committed:
Renamed the bot and implemented the skeleton structure

leontan2 committed:
Renamed the bot

leontan2 committed:
removed unneccessary code

leontan2 committed:
Improved the skeletal version of Duke so that it echos commands entered by the user, and exits when the user types the command bye.

leontan2 committed:
test

leontan2 committed:
Added the ability to store whatever text entered by the user and display them back to the user when requested.

leontan2 committed:
cleaned up unnecessary code

leontan2 committed:
Added the ability to mark tasks as done. Optionally, add the ability to change the status back to not done.

leontan2 committed:
Add support for tracking three types of tasks: TODOS, DEADLINES, EVENTS

leontan2 committed:
Used the input/output redirection technique to semi-automate the testing of Duke.

leontan2 committed:
Taught the chatbot to deal with errors such as incorrect inputs entered by the user.

leontan2 committed:
Added support for deleting tasks from the list.

leontan2 committed:
slight changes to the expected text for the automated I/O testing

leontan2 committed:
Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when the chatbot starts up.

leontan2 committed:
Teach the chatbot how to understand dates and times.

leontan2 committed:
Teach the chatbot how to understand dates and times.

leontan2 committed:
Revert "Teach the chatbot how to understand dates and times."

This reverts commit be2199ca09e8cea40c37177c1fd6835d98283103.

leontan2 committed:
Revert "Revert "Teach the chatbot how to understand dates and times.""

This reverts commit b1ac808fb0ef5c9b1251b908838ba25e258c543b.

leontan2 committed:
change duke text files.

leontan2 committed:
Refactor the code to extract out closely related code as classes.

leontan2 committed:
Organize the classes into suitable java packages.

leontan2 committed:
Organize the classes into suitable java packages.

leontan2 committed:
Merge branch 'branch-A-Packages'

leontan2 committed:
Merge branch 'add-gradle-support'

leontan2 committed:
Add JUnit tests

leontan2 committed:
Merge branch 'branch-A-JUnit'

leontan2 committed:
Package the app as an executable JAR file so that it can be distributed easily.

leontan2 committed:
Merge branch 'branch-A-Jar'

leontan2 committed:
Add JavaDoc comments to the code.

leontan2 committed:
Give users a way to find a task by searching for a keyword.

leontan2 committed:
Merge branch 'branch-Level-9'

leontan2 committed:
Add javaDocs to FindCommand

leontan2 committed:
Tweak the code to comply with a coding standard

leontan2 committed:
Merge branch 'branch-A-CodingStandard'

ChongWeiJie29 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,119 @@
+     * Checks if the AddCommand should trigger the program to exit.
+     *
+     * @return True if the command is an exit command, false otherwise.
+     */
```
Since this is an overriden method and the comment is similar to the parent class method, perhaps you can consider using the `@inheritDoc` tag?

ChongWeiJie29 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,119 @@
+                break;
+            default:
+                throw new DukeException(ui.messageCard("There may be wrong parameters inputted in, please check again!"));
+            }
```
I like the good use of switch statements here, as well as its proper formatting and indentation

ChongWeiJie29 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,119 @@
+
+    /**
+     * Constructor for creating an AddCommand.
+     *
```
Perhaps you may consider changing the phrasing to be "Constructs..." instead of the current "Constructor..."?

ChongWeiJie29 commented on src/test/java/duke/command/AddCommandTest.java:
```diff
@@ -0,0 +1,38 @@
+
+public class AddCommandTest {
+    @Test
+    public void InvalidDate() {
```
Would it help with clarity if you named it following the suggested structure "featureUnderTest_testScenario_expectedBehavior()" as per the coding standard for test cases?

ChongWeiJie29 commented on src/test/java/duke/task/TodoTest.java:
```diff
@@ -0,0 +1,17 @@
+    public void anotherDummyTest(){
+        assertEquals(4, 4);
+    }
+}
```
Overall, I think you did a very good job with the coding standard. There were barely any violations to point out. Everything is well formatted, named and documented

ChongWeiJie29 reviewed (commented):
Overall, coding standard expectations were well met. Everything was well documented and adhered to. Main comments were with regards to the phrasing of JavaDoc comments.

leontan2 committed:
Use checkStyle to detect coding style violations.

leontan2 committed:
Merge branch 'branch-A-CheckStyle'

leontan2 committed:
Add a GUI to the chatbot. Use the JavaFX technology to implement the GUI.

leontan2 committed:
Merge branch 'branch-Level-10'

Chen1x commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+package duke;
+
+import duke.command.Command;
```
I really like your abstraction and encapsulation of different commands. Good structure!

Chen1x commented on src/main/java/duke/Launcher.java:
```diff
@@ -0,0 +1,13 @@
+package duke;
+import duke.gui.Main;
+import javafx.application.Application;
```
I normally add a empty line between each imports from different source to make it tidier

Chen1x commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+package duke;
```
Good coding quality! LGTM

Chen1x commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,125 @@
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        try {
+            switch (category) {
```
Brilliant use of Switch here

Chen1x reviewed (commented):
LGTM!

leontan2 committed:
Create new jar file for the gui

leontan2 committed:
Add assertions to document important assumptions that should hold
at various points in the code.

In this commit, we've introduced assertions to improve input
validation and enhance the overall robustness of the Command class.

Changes made in this commit:
- Added assertions to check the validity of user input parameters in
  the Command classes and Parser class.
- Included clear error messages in assertions to provide developers
  with helpful information when an assertion fails.

These assertions help us catch potential programming errors early
in the development process, reducing the risk of unexpected
behaviour and improving code quality.

leontan2 committed:
Improve code quality

This commit focuses on critically examining the existing codebase
and making necessary refactoring changes to enhance code quality.
The changes made align closely with the 'Code Quality' best
practices and principles learned, ensuring that the codebase is
more maintainable, readable, and follows established coding
standards.

Let's
- Improve variable naming for clarity and consistency.
- Enhance code readability through meaningful comments and Javadoc.
- Streamline error handling and exception messages.
- Ensure adherence to established coding conventions.

By following these code quality improvements, the codebase is now
more robust and easier to maintain, contributing to better software
development practices.

leontan2 committed:
Merge pull request #1 from leontan2/branch-A-Assertions

Add assertions

leontan2 committed:
Merge branch 'master' of https://github.com/leontan2/ip

leontan2 committed:
Merge pull request #2 from leontan2/heads/branch-A-CodeQuality

Improve code quality

leontan2 committed:
Merge branch 'master' of https://github.com/leontan2/ip

leontan2 committed:
Set up Continuous Integration (CI)

Let's implement Continuous Integration (CI) using GitHub
Actions. The CI workflow is configured to automatically build and
test the project whenever changes are pushed to the repository.
This enhancement improves code quality and ensures that the project
remains in a consistent and functional state.

leontan2 committed:
Merge branch 'branch-A-CI'

leontan2 committed:
Implement 'Undo' Feature

This commit adds the 'Undo' feature to the Duke application, which
allows users to reverse the most recent command. This includes
adding, deleting, marking, or unmarking tasks.

Main Changes:
- Introduces the 'UndoCommand' class to handle undoing actions.
- Extends the 'Ui' class to store the most recent command and
task-related information.

leontan2 committed:
Merge pull request #3 from leontan2/branch-C-Undo

Implement 'Undo' Feature

leontan2 committed:
Merge branch 'master' of https://github.com/leontan2/ip

leontan2 committed:
Removed some lines from .yml files for CI

leontan2 committed:
Improve the GUI to make it more polished.

Changes made:
- Removed the unnecessary collapsing of the horizontal text which
replaces it with ellipses.

leontan2 committed:
Merge branch 'branch-A-BetterGui'

leontan2 committed:
Add a User Guide to the project by updating README.md

leontan2 committed:
Merge branch 'branch-A-UserGuide'

leontan2 committed:
Made Ui.png the right name

leontan2 committed:
Change the name to bob.jar

leontan2 committed:
Update README.md in docs
